### ISSUE TITLE
Make logger object name patterns configurable

### ISSUE THREAD

**Issue by lpulley** [astral-sh/ruff#12613]
Currently, Ruff uses some hard-coded name "patterns" to determine whether a module-local object might be a logger (for the purposes of e.g. `flake8-logging-format` G rules).

https://github.com/astral-sh/ruff/blob/fc16d8d04d86aa94a8aac14bbb87089b64d443a1/crates/ruff_python_semantic/src/analyze/logging.rs#L60-L74

We've found that we have some loggers called `_LOG` which aren't matched, and I was disappointed to find that these aren't configurable. Maybe this should be a setting list of regular expressions or globs or something like that?

**Comment by lpulley** [comment]
I suppose, at the very least, it would be nice to add `starts_with("_log")` and `starts_with("_LOG")`, but that seems like a band-aid.

**Comment by MichaReiser** [comment]
This makes sense to me, at least short-term. The long term goal is to use type inference to detect if a variable is a logger

**Comment by spaceone** [comment]
from duplicate: https://github.com/astral-sh/ruff/issues/20436

with config:
```toml
[tool.ruff.lint]
logger-objects = ['foo.get_it', 'AUTH']
```

The following use cases exists:
```python
from foo import get_it
get_it.info('foo %s' % (1,))   # detected

blah = get_it()
blah.info('foo %s' % (1,))  # not detected
get_it().info('foo %s' % (1,))  # not detected

blub = get_it
blub.info('foo %s' % (1,))  # also not detected

AUTH.info('foo %s' % (1,))  # also not detected
```

So, I would like logger-objects to be the result of a function/method call.
Furthermore I would like to specify the log variable names, our loggers are called `AUTH`, `NETWORK`, `ACL`, etc. and therefor don't match anything like `.*log.*`.
If I just put these names into logger-objects, they are ignored.

**Comment by a1tus** [comment]
Voting for this, we're using customer helpers/wrappers like `log_exc`, `log_err`, etc. everywhere, so we have to insert `noqa: BLE001` each time.

Would be nice to have ability to set some explicit names (or maybe paths) that are considered as logger functions.

**Comment by a1tus** [comment]
I've created PR to address this issue https://github.com/astral-sh/ruff/pull/23783

**Comment by quagly** [comment]
I always use `_log` for my module loggers.  So the short term hack of `starts_with("_log")` would cover all my use cases.
